src/cli/self_update/windows.rs

```diff
@@ -859,30 +845,28 @@ pub struct RegistryValueId {
 
 #[cfg(any(test, feature = "test"))]
 impl RegistryValueId {
-    pub fn get_value<T: FromRegValue>(&self) -> io::Result<Option<T>> {
-        self.get()?.map(|v| T::from_reg_value(&v)).transpose()
+    pub fn get_value(&self) -> Result<Option<Value>> {
+        self.get()
     }
 
-    fn get(&self) -> io::Result<Option<RegValue>> {
-        let sub_key = RegKey::predef(HKEY_CURRENT_USER)
-            .open_subkey_with_flags(self.sub_key, KEY_READ | KEY_WRITE)?;
-        match sub_key.get_raw_value(self.value_name) {
+    fn get(&self) -> Result<Option<Value>> {
+        let sub_key = CURRENT_USER.create(self.sub_key)?;
+        match sub_key.get_value(self.value_name) {
             Ok(val) => Ok(Some(val)),
-            Err(ref e) if e.kind() == io::ErrorKind::NotFound => Ok(None),
-            Err(e) => Err(e),
+            Err(e) if e.code() == HRESULT::from_win32(ERROR_FILE_NOT_FOUND) => Ok(None),
+            Err(e) => Err(e.into()),
         }
     }
 
-    pub fn set_value(&self, new: Option<impl ToRegValue>) -> io::Result<()> {
-        self.set(new.map(|s| s.to_reg_value()).as_ref())
+    pub fn set_value(&self, ty: Option<Type>, new: Option<&[u8]>) -> Result<()> {
+        self.set(ty, new)
     }
 
-    fn set(&self, new: Option<&RegValue>) -> io::Result<()> {
-        let sub_key = RegKey::predef(HKEY_CURRENT_USER)
-            .open_subkey_with_flags(self.sub_key, KEY_READ | KEY_WRITE)?;
-        match new {
-            Some(new) => sub_key.set_raw_value(self.value_name, new),
-            None => sub_key.delete_value(self.value_name),
+    fn set(&self, ty: Option<Type>, new: Option<&[u8]>) -> Result<()> {
```

Comment:
It looks like:
```rs
ty: Option<Type>, new: Option<&[u8]>,
```
... has introduced new input states that don't really make sense?

Would the following better convey what you intended to do?
```rs
new: Option<(Type, &[u8])>,
```